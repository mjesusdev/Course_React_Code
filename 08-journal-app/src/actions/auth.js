import { types } from "../types/types";
import { firebase, googleAuthProvider } from '../firebase/firebase-config';

// Sign in with Email & Password
export const startLoginEmailPassword = ( email, password ) => {
    return ( dispatch ) => {
        setTimeout(() => {
            dispatch( login(123, 'Peter') );
        }, 3000)
    }
}

// Register User with Email & Password
export const startRegisterUser = ( email, password, name ) => {
    return ( dispatch ) => {

        firebase.auth().createUserWithEmailAndPassword( email, password)
            .then( async ({ user }) => {

                await user.updateProfile({ displayName: name });

                dispatch(
                    login( user.uid, user.displayName )
                )
            })

            .catch( error => {
                console.log(error);
            })
    }
}

// Sign in with Google Account
export const startGoogleLogin = () => {
    return ( dispatch ) => {
        firebase.auth().signInWithPopup( googleAuthProvider )
            .then( ({ user }) => {
                dispatch(
                    login( user.uid, user.displayName )
                )
            });
    }
}

// Export
export const login = ( uid, displayName ) => ({
    type: types.login,
    payload: {
        uid,
        displayName
    }
})