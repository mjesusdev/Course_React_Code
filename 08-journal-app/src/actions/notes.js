import { db } from "../firebase/firebase-config";
import { loadNotes } from "../helpers/loadNotes";
import { types } from "../types/types";

export const startNewNote = () => {
    return async( dispatch, getState ) => {

        const { uid } = getState().auth;
        console.log(uid);

        const newNote = {
            title: '',
            body: '',
            date: new Date().getTime()
        }

        const document = await db.collection(`${ uid }/journal/notes`).add( newNote );

        dispatch( activeNote( document.id, newNote ) );
    }
}

export const activeNote = ( id, note ) => ({
    type: types.notesActive,
    payload: {
        id,
        ...note
    }
});

export const startLoadingNotes = ( uid ) => {
    return async( dispatch ) => {
        const notes = await loadNotes( uid );
        await dispatch( setNotes(notes) );
    }
}

export const setNotes = ( notes ) => ({
    type: types.notesLoad,
    payload: notes
});