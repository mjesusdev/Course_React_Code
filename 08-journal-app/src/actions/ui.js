import { types } from '../types/types';

export const setError = ( err ) => ({
    type: types.uiSetError,
    payload: err
});

export const removeError = ( err ) => ({
    type: types.uiRemoveError,
    payload: err
});

export const startLoading = () => ({
    type: types.uiStartLoading
})

export const finishLoading = () => ({
    type: types.uiFinishLoading
})