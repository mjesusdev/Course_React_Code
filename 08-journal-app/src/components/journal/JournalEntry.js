import React from 'react';

export const JournalEntry = () => {
    return (
        <div className="journal__entry pointer">
            <div 
                className="journal__entry-picture"
                style={{
                    backgroundSize: 'cover',
                    backgroundImage: 'url(https://www.nexofin.com/archivos/2017/08/soltodo-el-dia_88d9c462.jpg)'
                }}
            ></div>

            <div className="journal__entry-body">
                <p className="journal__entry-title">
                    A new day ðŸ’ª
                </p>
                <p className="journal__entry-content">
                    Lorem ipsum dolor sit amet, consectetuer adipiscing elit
                </p>
            </div>

            <div className="journal__entry-date-box">
                <span>Monday</span>
                <h4>28</h4>
            </div>

        </div>
    )
}
