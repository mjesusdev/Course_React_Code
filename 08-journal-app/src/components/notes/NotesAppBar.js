import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote } from '../../actions/notes';

export const NotesAppBar = () => {
    
    const dispatch = useDispatch();
    const { active } = useSelector( state => state.notes );

    const handleSave = () => {
        dispatch( startSaveNote(active) );
    }

    return (
        <div className="notes__appbar">
            <span>{ Date() }</span>

            <div>
                <button className="btn">
                    Picture
                </button>

                <button 
                    className="btn"
                    onClick={ handleSave }
                >
                    Save
                </button>
            </div>
        </div>
    )
}
