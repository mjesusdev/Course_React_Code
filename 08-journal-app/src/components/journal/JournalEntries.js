import React from 'react';
import { JournalEntry } from './JournalEntry';

export const JournalEntries = () => {

    const entries = [1,2,3,4,5,6,7,8,9,10];

    return (
        <div className="journal__entries">
            
            {
                entries.map( value => (
                    <JournalEntry key={ value }/>
                ))
            }

        </div>
    )
}
