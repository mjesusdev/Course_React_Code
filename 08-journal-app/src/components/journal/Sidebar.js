import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { JournalEntries } from './JournalEntries';
import { startLogout } from '../../actions/auth';

export const Sidebar = () => {

    const dispatch = useDispatch();
    // Get user logged in the app
    const { name } = useSelector( state => state.auth );

    const handleLogout = () => {
        dispatch( startLogout() );
    }

    return (
        <aside className="journal__sidebar">
            
            <div className="journal__sidebar-navbar">
                <h3 className="mt-5">
                    <i className="far fa-moon"></i>
                    <span> { name } </span>
                </h3>

                <button 
                    className="btn"
                    onClick={ handleLogout }
                >
                    Logout
                </button>
            </div>

            <div className="journal__new-entry">
                <i className="far fa-calendar-plus fa-5x"></i>
                <p className="mt-5">
                    New entry
                </p>
            </div>

            <JournalEntries />

        </aside>
    )
}
