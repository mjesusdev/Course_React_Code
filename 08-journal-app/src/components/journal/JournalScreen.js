import React from 'react';
import { useSelector } from 'react-redux';

import { Sidebar } from './Sidebar';
import { NoteScreen } from '../notes/NoteScreen';
import { NothingSelected } from './NothingSelected';

export const JournalScreen = () => {

    const { active } = useSelector( state => state.notes );

    console.log(active);

    return (
        <div className="journal__main-content">
            <Sidebar />

            <main>
                {
                    ( active )
                        ? ( <NoteScreen /> )
                        : ( <NothingSelected /> )
                }
            </main>
        </div>
    )
}
