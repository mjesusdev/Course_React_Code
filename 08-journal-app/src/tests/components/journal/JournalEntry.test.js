import React from 'react';

import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';

const middlewars = [thunk];
const mockStore = configureStore(middlewars);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
    id: 10,
    date: 0,
    title: 'Hello',
    body: 'World',
    url: 'https://localhost/cualquier/cosa.jpg'
}

const wrapper = mount(
    <Provider store={ store }>
        <JournalEntry { ...note } /> 
    </Provider>
);

describe('Test in <JournalEntry />', () => {

    test('should show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should active the note', () => {
        wrapper.find('.journal__entry').simulate('click');

        expect( store.dispatch ).toHaveBeenCalledWith(
            activeNote( note.id, { ...note } )
        );
    });
    
})
