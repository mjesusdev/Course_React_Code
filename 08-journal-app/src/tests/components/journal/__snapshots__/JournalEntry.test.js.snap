// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <JournalEntry /> should show correctly 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate_faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://localhost/cualquier/cosa.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hello
    </p>
    <p
      className="journal__entry-content"
    >
      World
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
       
      Thursday
       
    </span>
    <h4>
       
      1st
       
    </h4>
  </div>
</div>
`;
